### DISCUSSION TITLE
Zwift remote streaming on Sway with Sunshine and Moonlight

### DISCUSSION THREAD

**Discussion by quietvoid** [netbrain/zwift#49] [Show and tell]
This is my setup for streaming Zwift from my desktop PC over to Moonlight on any supported device (I use a Linux laptop).
It's very Sway specific due to the script creating a virtual wlroots output and moving Zwift.

Sunshine application configuration:
- Command preparations:
  - Do: `sudo systemctl start docker`, Undo: `sudo systemctl stop docker.service docker.socket`
    Optional if Docker is already always running.
- Command: `zsh zwift-sunshine.sh`

`zwift-sunshine.sh`
```zsh
#!/usr/bin/zsh
set -e

function set_zwift_output {
	ZWIFT_OUTPUT=$(swaymsg -t get_outputs | jq -r '.[] | select(.name | startswith("HEADLESS")) | .name')
}

set_zwift_output
if [[ -z $ZWIFT_OUTPUT ]]; then
	swaymsg create_output
	set_zwift_output
fi

function zwift_cleanup {
	echo "Stopping zwift container"
	docker stop zwift || true
	swaymsg output $ZWIFT_OUTPUT unplug
}

trap zwift_cleanup INT QUIT TERM EXIT

zsh zwift.sh
sleep 5

swaymsg "[title=\"Zwift\"]" move output $ZWIFT_OUTPUT
sleep 1
swaymsg "[title=\"Zwift\"]" focus
swaymsg "[title=\"Zwift\"]" fullscreen

docker wait zwift
```
One can replace `zsh zwift.sh` with the proper `zwift.sh` path (if it's not in PATH).

Behaviour:
- Creates a virtual 1920x1080 output if not existing already.
- Starts Zwift with a `zwift` named container (may require `zwift.sh` adjustments).
- Moves the Zwift window to the virtual output, focuses it and fullscreens.
- Keeps running until Zwift is closed (by waiting for the container to stop).

In a Moonlight client, the game starts and shows up. When exiting Zwift, the stream ends automatically.

**Comment by mohdyusuf** [comment]
Interesting!! you just gave me an idea on how to stream zwift from my PC to my treadmill :)
